src/components/hoc/withContextProvider.tsx

```diff
@@ -0,0 +1,11 @@
+import React from 'react'
+
+export function withContextProvider<T>(getValue: () => T): [React.FunctionComponent<{ children: React.ReactNode }>, React.Context<T | undefined>] {
```

Comment:
This isn't really a HOC, since it takes a callback function, not a component. It needs another name and another home.

I am also worried about updating the context value on every render - I think this will force a re-render of each linked child. According to the docs, React uses `===` on the context value to decide whether or not an update has occurred. Using a raw `return {...}` will never be `===` equal, so the update will always take place.

So, if we want to implement these contexts correctly, we need to do something like this:

```ts
export const SceneDrawerContext = React.createContext(undefined)

export function SceneDrawerProvider(props: { children: React.ReactNode }) {
  const [isRatioDisabled, setIsRatioDisabled] = React.useState(false)
  const drawerHeight = useSharedValue(0)
  const drawerOpenRatio = useSharedValue(1)
  const drawerOpenRatioStart = useSharedValue(1)

  const value = React.useMemo(
    () => ({
      drawerHeight,
      drawerOpenRatio,
      drawerOpenRatioStart,
      isRatioDisabled,
      setIsRatioDisabled
    }),
    [drawerHeight, drawerOpenRatio, drawerOpenRatioStart, isRatioDisabled]
  )

  return <SceneDrawerContext.Provider value={value}>{props.children}</SceneDrawerContext.Provider>
}
```

I'm not sure if your `withContextProvider` helper function adds much, compared to just doing things directly as above. Maybe you can refactor your helper to somehow help with the memorization, but I don't have a clear idea about what this would look like.

Another idea is to combine `makeUseContextValue` and `withContextProvider` into a single step - `createStateProvider`. This would reflect the `StateProvier` nomenclature used later, and will create both the hook and the provider in one step.